File: seleniumPHP/OwnerTest.php
<?php
require 'vendor/autoload.php';

use Behat\Mink\Mink,
    Behat\Mink\Session,
    Behat\Mink\Driver\Selenium2Driver;

use Selenium\Client as SeleniumClient;

/**
*
*/
class OwnerTest extends PHPUnit_Framework_TestCase
{
    public function testSaveSuccessfull()
    {
        $url = 'http://localhost:9966/petclinic';
        $browser = 'firefox';

        $driver = new \Behat\Mink\Driver\Selenium2Driver( 'firefox', 'base_url');

        // init session:
        $session = new \Behat\Mink\Session($driver);

        // start session:
        $session->start();

        // open some page in browser:
        $session->visit($url);

        //navigate through some pages
        $page = $session->getPage();
        $page->findLink('Find owners')->click();
        $page->findLink('Add Owner')->click();

        //complete a form
        $page->findField('firstName')->setValue('My Name');
        $page->findField('lastName')->setValue('My Last Name');
        $page->findField('city')->setValue('My City');
        $page->findField('address')->setValue('My Address');
        $page->findField('telephone')->setValue('123456');

        $page->find('css', '.form-actions button')->click();

        $title = $page->find('css', 'div.container h2')->getText();

        $this->assertEquals('Owner Information', $title);

        //close browser
        $session->stop();
    }
}

File: selenium-petclinic/src/test/java/org/webtesting/functional/OwnerTest.java
package org.webtesting.functional;

import org.hamcrest.Matcher;
import org.junit.Assert;
import org.junit.Test;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.webtesting.pages.EditOwner;
import org.webtesting.pages.FindOwners;
import org.webtesting.pages.HomePage;
import org.webtesting.pages.Menu;

public class OwnerTest {

    @Test
    public void shouldCreateOwner() {
        WebDriver webDriver = new FirefoxDriver();

        webDriver.get("http://localhost:9966/petclinic");
        //webDriver.get("http://petclinic.cloudapp.net");

        WebElement link = webDriver.findElement(By.linkText("Find owners"));
        link.click();

        WebElement addOwner = webDriver.findElement(By.linkText("Add Owner"));
        addOwner.click();

        WebElement firstName = webDriver.findElement(By.id("firstName"));
        firstName.sendKeys("My Name");

        WebElement lastName = webDriver.findElement(By.id("lastName"));
        lastName.sendKeys("My Last Name");

        WebElement address = webDriver.findElement(By.id("address"));
        address.sendKeys("My Address");

        WebElement city = webDriver.findElement(By.id("city"));
        city.sendKeys("My City");

        WebElement telephone = webDriver.findElement(By.id("telephone"));
        telephone.sendKeys("123456");

        WebElement submit = webDriver.findElement(By.cssSelector(".form-actions button"));
        submit.click();

        WebElement title = webDriver.findElement(By.cssSelector("div.container h2"));

        Assert.assertEquals("Owner Information", title.getText());

        webDriver.quit();
    }

    @Test
    public void shouldNotCreateOwnerByPhoneNumberValidation() {
        WebDriver webDriver = new FirefoxDriver();

        //webDriver.get("http://petclinic.cloudapp.net");
        webDriver.get("http://localhost:9966/petclinic");

        WebElement link = webDriver.findElement(By.linkText("Find owners"));
        link.click();

        WebElement addOwner = webDriver.findElement(By.linkText("Add Owner"));
        addOwner.click();

        //Formulario
        WebElement firstName = webDriver.findElement(By.id("firstName"));
        firstName.sendKeys("My Name");

        WebElement lastName = webDriver.findElement(By.id("lastName"));
        lastName.sendKeys("My Last Name");

        WebElement address = webDriver.findElement(By.id("address"));
        address.sendKeys("My Address");

        WebElement city = webDriver.findElement(By.id("city"));
        city.sendKeys("My City");

        WebElement telephone = webDriver.findElement(By.id("telephone"));
        telephone.sendKeys("abc");

        WebElement submit = webDriver.findElement(By.cssSelector(".form-actions button"));
        submit.click();

        WebElement title = webDriver.findElement(By.cssSelector("div.container h2"));

        Assert.assertEquals("New Owner", title.getText());

        WebElement errorMessage = webDriver.findElement(By.cssSelector(".error span"));

        Assert.assertEquals("numeric value out of bounds (<10 digits>.<0 digits> expected)", errorMessage.getText());

        webDriver.quit();
    }

    
}
